mapie/classification.py

```diff
@@ -658,95 +859,147 @@ def predict(
         (n_samples,) and (n_samples, n_classes, n_alpha) if alpha is not None.
         """
         # Checks
+        cv = self._check_cv(self.cv)
         include_last_label = self._check_include_last_label(include_last_label)
         alpha_ = check_alpha(alpha)
         check_is_fitted(self, self.fit_attributes)
         if self.image_input:
             check_input_is_image(X)
         X = check_array(
-            X, force_all_finite=False, ensure_2d=self.image_input,
-            allow_nd=self.image_input, dtype=["float64", "object"]
+            X,
+            force_all_finite=False,
+            ensure_2d=self.image_input,
+            allow_nd=self.image_input,
+            dtype=["float64", "object"]
         )
+
+        # Estimate probabilities from estimator(s)
         y_pred = self.single_estimator_.predict(X)
-        y_pred_proba = self.single_estimator_.predict_proba(X)
-        y_pred_proba = self._check_proba_normalized(y_pred_proba)
+        if cv == "prefit":
+            y_pred_proba = self.single_estimator_.predict_proba(X)
+            y_pred_proba = self._check_proba_normalized(y_pred_proba)
+        else:
+            y_pred_proba_k = Parallel(
+                n_jobs=self.n_jobs, verbose=self.verbose
+            )(
+                delayed(self._predict_oof_model)(est, X)
+                for est in self.estimators_
+            )
+            if agg_scores == "crossval":
+                y_pred_proba = np.stack(
+                    [y_pred_proba_k[k] for k in self.k_], axis=0
```

Comment:
Is it a loop on `n_samples_train` ? what is the shape of `y_pred_proba` at this step ?

Reply:
Changed to `y_pred_proba_k[self.k_]`